File: lab/python/arquivos/estatis_de_arquivo.py
def ler_texto(caminho) -> str:
    with open(caminho, "r", encoding="utf-8") as file:
        return file.read()


def contar_caracteres(texto):
    return len(texto)


def contar_linhas(texto):
    return len(texto.splitlines())


def contar_palavras(texto):
    return len(texto.split())


def main():
    caminho = input("Digite o caminho do arquivo: ")
    try:
        texto = ler_texto(caminho)
    except FileNotFoundError:
        print("Erro: Arquivo não encontrado.")
        return
    except PermissionError:
        print("Erro: Permissão negada.")
        return

    print(f"Caracteres: {contar_caracteres(texto)}")
    print(f"Linhas: {contar_linhas(texto)}")
    print(f"Palavras: {contar_palavras(texto)}")


if __name__ == "__main__":
    main()

File: lab/javascript/arquivos/estatis_de_arquivo.js
const fs = require("fs");

function lerTexto(caminho) {
    try {
        return fs.readFileSync(caminho, "utf-8");
    } catch (err) {
        if (err.code === "ENOENT") {
            console.log("Erro: arquivo não encontrado.");
        } else if (err.code === "EACCES") {
            console.log("Erro: permissão negada.");
        } else {
            console.log("Erro desconhecido:", err.message);
        }
        process.exit(1);
    }
}

function contarCaracteres(texto) {
    return texto.length;
}

function contarLinhas(texto) {
    return texto.split(/\r?\n/).length;
}

function contarPalavras(texto) {
    return texto.trim().split(/\s+/).filter(Boolean).length;
}

function main() {
    const caminho = require("prompt-sync")()("Digite o caminho do arquivo: ");
    const texto = lerTexto(caminho);

    console.log(`Caracteres: ${contarCaracteres(texto)}`);
    console.log(`Linhas: ${contarLinhas(texto)}`);
    console.log(`Palavras: ${contarPalavras(texto)}`);
}

main();
